leader-election-impl/src/main/java/com/palantir/leader/proxy/AwaitingLeadershipProxy.java

```diff
@@ -234,14 +239,21 @@ protected Object handleInvocation(Object proxy, Method method, Object[] args) th
 
         if (!method.getReturnType().equals(ListenableFuture.class)) {
             Object delegate = AtlasFutures.getUnchecked(delegateFuture);
-            try {
+            try (CloseableTracer ignored = CloseableTracer.startSpan("execute-on-delegate")) {
                 return method.invoke(delegate, args);
             } catch (InvocationTargetException e) {
                 throw handleDelegateThrewException(leadershipToken, e);
             }
         } else {
             return FluentFuture.from(delegateFuture)
-                    .transformAsync(delegate -> (ListenableFuture<Object>) method.invoke(delegate, args),
+                    .transformAsync(delegate ->
+                                    Tracers.wrapListenableFuture("execute-on-delegate-async", () -> {
+                                        try {
+                                            return (ListenableFuture<Object>) method.invoke(delegate, args);
+                                        } catch (IllegalAccessException | InvocationTargetException e) {
+                                            return Futures.immediateFailedFuture(e);
```

Comment:
Do we not want to handle the `IAE` below? not that I think it'll come up that often, I don't think we handled it previously anyway, but seems a bit odd, and it'll come out as a 500 iirc.

Reply:
unclear, if it's unhandled before, and we haven't really seen instances of it, I'll leave it up to you if you want to change it, it just caught my eye with the try-catch in the new diff but was surprised it was missing from the old diff, not fussed either way